src/order/order_iso_nat.lean

```diff
@@ -43,3 +45,28 @@ theorem well_founded_iff_no_descending_seq [is_strict_order α r] :
     by { rw [function.iterate_succ'], apply h }⟩⟩⟩
 
 end rel_embedding
+
+namespace partial_order
+
+variables (α : Type*) [partial_order α]
```

Comment:
Should the `satisfies_acc` class assume only `has_le`? Or `preorder`? `partial_order` seems "overkill".

Reply:
Agreed, now done.